Application/Validation/NameValidator.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Core.Dto;
+using Core.Dto.Request;
+using FluentValidation;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Application.Validation
+{
+    public class NameValidator : AbstractValidator<NameDto>
+    {
+        public NameValidator(GeoLocationValidator geoLocationValidator)
+        {
+            RuleFor(x => x.Name)
+            .NotEmpty().WithMessage("Name is required.");
+
+            RuleFor(x => x.Meaning)
+                .NotEmpty().WithMessage("Meaning is required.");
+
+            RuleForEach(x => x.Etymology)
+                .SetValidator(new EtymologyValidator());
+
+            RuleForEach(x => x.Videos)
+                .SetValidator(new EmbeddedVideoValidator());
+
+            RuleForEach(x => x.GeoLocation)
+                .SetValidator(geoLocationValidator);
+
+            RuleFor(x => x.FamousPeople)
```

Comment:
The `FamousPeople` and `Syllables` fields are not required. The validations should only apply when a value is supplied.

Reply:
resolved